addon/wolfcrypt/main.cpp

```diff
@@ -0,0 +1,149 @@
+#include <napi.h>
+#include <stdio.h>
+#include <cstring>
+#include "./h/evp.h"
+#include "./h/hmac.h"
+#include "./h/rsa.h"
+#include "./h/sha.h"
+#include <wolfssl/options.h>
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/aes.h>
+
+using namespace Napi;
+
+typedef struct WrappedKey
+{
+  Aes aes[1];
+  uint8_t key[32];
+  uint8_t iv[16];
+} WrappedKey;
+
+Napi::Number MakeAes(const Napi::CallbackInfo& info)
+{
+  int ret = 0;
+  Napi::Env env = info.Env();
+  WrappedKey* key = (WrappedKey*)( info[0].As<Napi::Uint8Array>().Data() );
+
+  memcpy( key->key, ( info[1].As<Napi::Uint8Array>().Data() ), 32 );
+  memcpy( key->iv, ( info[2].As<Napi::Uint8Array>().Data() ), 16 );
+
+  //ret = wc_AesInit( key->aes, NULL, INVALID_DEVID );
```

Comment:
Why is `wc_AesInit` not being called? Is `key->key` guaranteed to be zero initialized?